streaming/base/stream.py

```diff
@@ -140,11 +141,31 @@ def __init__(self,
 
         self.validate_hash = validate_hash
 
+        if local is None:
+            self.local = self._get_temporary_directory()
+            if get_local_rank() == 0:
```

Comment:
why is this here? if i am not mistaken, `self.local` will get created incidentally when local rank zero downloads its index

Reply:
I think we want to make sure that if multiple streaming datasets are created, they use different local directories instead of all being instantiated with None so we require that the temporary directory doesn't exist beforehand.